tests/src/psa_exercise_key.c

```diff
@@ -733,13 +736,28 @@ psa_status_t mbedtls_test_psa_raw_key_agreement_with_self(
                     PSA_RAW_KEY_AGREEMENT_OUTPUT_MAX_SIZE);
     }
 
+    psa_set_key_type(&shared_secret_attributes, PSA_KEY_TYPE_DERIVE);
+    psa_set_key_usage_flags(&shared_secret_attributes, PSA_KEY_USAGE_DERIVE | PSA_KEY_USAGE_EXPORT);
+
+    status = psa_key_agreement(key, public_key, public_key_length, alg,
+                               &shared_secret_attributes, &shared_secret_id);
+
+    if (key_destroyable && status == PSA_ERROR_INVALID_HANDLE) {
```

Comment:
Proposal for (potential) improvement: if the check on line 726 is OK, i.e. we jump into the check
`if (key_destroyable && status == PSA_ERROR_INVALID_HANDLE) {`
after `psa_raw_key_agreement`, why then jumping directly to `exit`, instead of trying also `psa_key_agreement`?

I see that this is pre-existing, since `goto exit` is used in all the cases when `(key_destroyable && status == PSA_ERROR_INVALID_HANDLE)`, but I think it wouldn't hurt to extensively test that the returned value is really `PSA_ERROR_INVALID_HANDLE` when the key has been destroyed before. Wdyt?

Reply:
I have been thinking over this for a bit. The problem here is that in the multi-threaded tests, it is possible for the passed in key or any key stored in a key slot to get deleted from under us. If the passed in key has been deleted, then none of the later tests will pass either way (you are now passing an empty slot).

If the export of the public key fails, then likewise nothing much else is going to work.

The problem with testing for `PSA_ERROR_INVALID_HANDLE` is, we don't necessarily know when the key has been deleted, it will have been done by another thread running tests, and we don't do communication between the threads in these tests as it would obviously be extremely complicated and probably ruin the initial point of the tests which is attempting to find race conditions with TSAN.

I'm still thinking about it, but I'm sceptical at this point we could do much better - these issues should only come in one run of the tests, which is the multithreaded run, which is only one of many runs of these tests.

Reply:
> I'm still thinking about it, but I'm sceptical at this point we could do much better

No worries, mine was just a proposal not an hard requirement ;). I'm absolutely OK with the current status